### PR TITLE
[clang][Interp] Add explicit dummy descriptors

### PR DESCRIPTION
**Pull request by tbaederr**
Instead of (ab)using incomplete array types for this, add a 'Dummy' bit to Descriptor. We need to be able to differentiate between the two when adding an offset.

### CODE REVIEW

**Comment by tbaederr** [comment]
Ping

**Comment by tbaederr** [comment]
Ping

File: clang/lib/AST/Interp/Descriptor.h [hunk 1]
```diff
@@ -109,6 +109,8 @@ struct Descriptor final {
   const bool IsTemporary = false;
   /// Flag indicating if the block is an array.
   const bool IsArray = false;
+  /// Flag indicating if this is a dummy descriptor.
+  const bool IsDummy = false;
```
Comment (AaronBallman) [inline clang/lib/AST/Interp/Descriptor.h:113]: Should we be considering using a bit-field to pack these together at some point? (As an NFC follow-up)

**Review by AaronBallman — APPROVED** [review]
LGTM

File: clang/lib/AST/Interp/Descriptor.h (hunk 1 above)
Comment (tbaederr) [reply clang/lib/AST/Interp/Descriptor.h:113]: Sure, I'll make a note.

**Pull request merged**